### ISSUE TITLE
Use Differential Evolution as tf keras optimizer.

### ISSUE THREAD

**Issue by ZeroMaxinumXZ** [tensorflow/probability#455]
Hi, so I'm looking to use the differential evolution optimizer, however it seems like most of these optimizers aren't really like 'normal' tf.train.Optimizers...  So, how should I go about using these optimizers as a Keras optimizer? Thanks.

**Comment by saxena-ashish-g** [comment]
Looking at Keras optimizer interface, it should be relatively straightforward to wrap the [tfp.optimizer.differential_evolution_one_step](https://github.com/tensorflow/probability/blob/master/tensorflow_probability/python/optimizer/differential_evolution.py#L121)  to implement the [optimizer.get_updates](https://github.com/keras-team/keras/blob/master/keras/optimizers.py#L85) method.

**Comment by ryankashi** [comment]
Hi! I am also interested in using the differential evolution optimizer inside of my tensorflow model, however I am somewhat confused by the comment above.

How exactly would I use the tfp de optimizer to create gradients inside of tensorflow model? For example, if I would like to train the weights inside of a tf.keras.layers.Dense layer, how would I do this using the tfp optimizer?

**Comment by w601sxs** [comment]
+1 ^

**Comment by cko22** [comment]
+1

**Comment by firatsarlar** [comment]
I was hoping to try same thing, and searching for starting point.
Currently I'm planning to check source code of to see how they plug their optimizer, and how can I do that with diff. evo. opt.
https://pygad.readthedocs.io/en/latest/README_pygad_kerasga_ReadTheDocs.html
I need some time to test it, but please share if you figure out a way to plug optimizer. ( I'll do if I can )
